### DISCUSSION TITLE
Timeouts for Test Runner?

### DISCUSSION THREAD

**Discussion by sernaferna** [nodejs/help#4666] [Q&A] [answered]
There's an ability to set a timeout at the `it`/`test` level, as well as at the `describe` level, but is there also a timeout that happens at the **file level**?

I'm coming from the Jest world where I'm used to using `it.each` (or even `describe.each`) when I want to execute a lot of similar tests. (i.e. execute the same code numerous times with different inputs with expected outputs.)

I've tried doing the same with `node:test` by having a `describe` that loops through numerous `test` calls, as well as looping through `describe` calls with "normal" calls to `test`, but when I involve things like in-memory databases for testing APIs and the setup required for all of this, I'm seeing too many timeouts. All of my tests might be within 2 seconds (for example), but by the time you add them all up, add in any setup happening in `before`, it can push the overall `describe` call beyond the 60s limit. But even if you break it down into multiple `describe` calls there still seems to be an overall 60s limit for the `.test.ts` file, so when I have a lot of tests I want to execute (it's difficult to nail down the exact limit) I have to create multiple files and split the code across them. 

If I'm right, and there's a file-level timeout, it would be good to call out in the documentation:

- Regardless of setting the `timeout` setting at the `test`/`it` level, is there a limit at the `describe` level? i.e. even if every `test` is within its own limits, `describe` has an overall timeout within which all tests must execute?
- Regardless of setting the `timeout` setting at the `describe` level, is there a limit at the **file** level, where even if there are multiple `describe` calls that are all within their own limits, the file itself will also time out at some point?

**Comment by MoLow** [comment, marked as answer]
`--test-timeout` should effect test files as well

**Reply by sernaferna** [reply]
Thanks for this!

Can you confirm if the default timeout limit at the file level is also 60s? So if I have multiple `describe` calls in one file, they _all_ have to complete within the 60s (unless the `--test-timeout` value is passed to the execution)?
